### ISSUE TITLE
`almost_swapped` false positive?

### ISSUE THREAD

**Issue by rjzak** [rust-lang/rust-clippy#10721]
### Description

Code which receives a parameter with the `ref` keyword claimed by clippy to be a swap.

Ref: https://github.com/profianinc/drawbridge/issues/423

Example:
```rust
pub async fn query(
    Extension(ref store): Extension<Arc<Store>>,
    ref cx: RepositoryContext,
    req: Request<Body>,
) -> impl IntoResponse { ..... }
```

Claims that `ref cx: RepositoryContext,` is a swap.

### Version

```text
rustc 1.69.0 (84c898d65 2023-04-16)
binary: rustc
commit-hash: 84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc
commit-date: 2023-04-16
host: x86_64-unknown-linux-gnu
release: 1.69.0
LLVM version: 15.0.7
```

**Comment by lukaslueg** [comment]
I don't quite see why TOPLEVEL_REF_ARG doesn't fire here!?

**Comment by rjzak** [comment]
Seems to be fixed by https://github.com/rust-lang/rust-clippy/pull/10499. We're no longer having this issue.
